### PR TITLE
test: Add b/e unit tests for displaying inquiries table

### PR DESCRIPTION
**Pull request by ilhanbae**
# Task
Add unit tests for listing inquiries.

[Link to Ticket](https://incredihire-academy.atlassian.net/jira/software/c/projects/SR/boards/3?selectedIssue=SR-47)

## Author Tasks

I added backend unit tests for `get_inquiry` and `get_inquiries` API / service using pytest. The tests are divided into two layers:
* service tests check if DB persists with the changes as we intended
* api tests check if API returns correct payload and status code 

I noticed that that when a test creates entries in database, those entries are not cleared afterward, which could cause issue with other tests. To address this, I added a pytest fixture to clean up specified tables after each test function. Currently, this fixture only clears the Inquiry table, but additional tables can be included as needed.

## Acceptance Criteria
_API_
- [x] should return error message with 404 status code when inquiry is not found
- [x] should return error message with 400 status code when invalid param is passed
- [x] should return single matching inquiry json object with 200 status code when single inquiry is requested  
- [x] should return list of inquiry json object with 200 status code when all inquiries are requested

_Service_
- [x] should return nothing when inquiry is not found
- [x] should return matching inquiry instance when single inquiry is requested
- [x] should return list of all inquiry instances when all inquiries are requested

### CODE REVIEW

File: backend/app/services/inquiries.py [hunk 1]
```diff
@@ -19,7 +19,9 @@ def get_inquiry_by_text(*, session: Session, text: str) -> Inquiry | None:
     return session_text
 
 
-def get_inquiry_by_id(*, session: Session, inquiry_id: uuid.UUID) -> Inquiry | None:
+def get_inquiry_by_id(*, session: Session, inquiry_id: UUID) -> Inquiry | None:
+    if not isinstance(inquiry_id, UUID):
```
Comment (jakethrasher) [inline backend/app/services/inquiries.py:23]: I'm not sure this check is necessary. Also we may migrate to a different class of identifier.

File: backend/app/tests/services/inquiry/test_get_inquiry_service.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    db.add(single_inquiry)
+    db.commit()
+    return single_inquiry
+
+
+def test_getInquiryService_whenCalledWithInvalidId_shouldRaiseException(
+    db: Session,
+) -> None:
+    with pytest.raises(ValueError, match=r"Id must be a valid UUID"):
+        get_inquiry_by_id(session=db, inquiry_id=12345)  # ruff: noqa
```
Comment (jakethrasher) [inline backend/app/tests/services/inquiry/test_get_inquiry_service.py:22]: Related to my comment in `services/inquiries.py` `get_inquiry_by_id` method. I think testing the type of the inquiry_id is not necessary so long as we are passing in a valid id (an id that maps to an inquiry in the db).

Comment (ilhanbae) [reply backend/app/tests/services/inquiry/test_get_inquiry_service.py:22]: My intention was to check the case when the `get_inquiry_by_id` service was called with wrong `inquiry_id` type from the API. I think I can enforce API to always call the service with valid param types - I'll add this test on the API layer.

Ruff linter was also catching invalid type when I was intentionally passing wrong typed param in the test. Trying to tell Ruff to ignore it for this test seemed to not work.

Removing this test from service layer.

File: backend/app/services/inquiries.py (hunk 1 above)
Comment (ilhanbae) [reply backend/app/services/inquiries.py:23]: Removing type check as suggested.

File: backend/app/tests/services/inquiry/test_get_inquiry_service.py (hunk 2 above)
Comment (jakethrasher) [reply backend/app/tests/services/inquiry/test_get_inquiry_service.py:22]: In the API layer, doesnt pydantic validate the type?

Comment (ilhanbae) [reply backend/app/tests/services/inquiry/test_get_inquiry_service.py]: You are right. Looks like the API is already configured to return an error response when passed with invalid param, effectively preventing it from calling the service:
```json
{
  "detail": [
    {
      "type": "uuid_parsing",
      "loc": [
        "path",
        "inquiry_id"
      ],
      "msg": "Input should be a valid UUID, invalid length: expected length 32 for simple format, found 1",
      "input": "1",
      "ctx": {
        "error": "invalid length: expected length 32 for simple format, found 1"
      }
    }
  ]
}
```

In the API layer test, I will just check response code and the error message.

**Review by jakethrasher — APPROVED** [review]
I pulled down the code and verified that all tests are passing. Looks good, great job

**Review by mcverter — APPROVED** [review]

**Pull request merged**